hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java

```diff
@@ -106,16 +125,46 @@ public Void visit(SqlIdentifier id) {
         return null;
     }
 
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
     @Override
     public Void visit(SqlDataTypeSpec type) {
-        throw error(type, RESOURCE.custom("Type specification is not supported"));
+        if (!(type.getTypeNameSpec() instanceof SqlBasicTypeNameSpec)) {
+            throw error(type, RESOURCE.custom("Complex type specifications are not supported"));
+        }
+
+        SqlTypeName typeName = SqlTypeName.get(type.getTypeName().getSimple());
+        switch (typeName) {
+            case BOOLEAN:
+            case TINYINT:
+            case SMALLINT:
+            case INTEGER:
+            case BIGINT:
+            case DECIMAL:
+            case REAL:
+            case DOUBLE:
+            case VARCHAR:
+            case ANY:
+            case NULL:
+            case TIME:
```

Comment:
Do we really need to handle temporal types in this PR, provided that they are not supported?

Reply:
fixed in 145a6e265125f2fa894d96377c32d9400d27b557